jujugui/static/gui/src/app/components/deployment/summary/summary.js

```diff
@@ -43,6 +43,59 @@ YUI.add('deployment-summary', function() {
       validateForm: React.PropTypes.func.isRequired
     },
 
+    getInitialState: function() {
+      return {
+        regions: [],
+        activeRegion: this.props.deploymentStorage.region
+      };
+    },
+
+    componentWillMount: function() {
+      // If the model has already been deployed then we already
+      // have all the information we need.
+      var props = this.props;
+      if (props.modelCommitted) { return; }
+      // Because the user may have selected a template from the choose-cloud
+      // component we won't have any of the location data for the template
+      // to be able to properly select the correct region or cloud from the
+      // drop down list.
+      var deploymentStorage = props.deploymentStorage;
+      var templateName = deploymentStorage.templateName;
+      this._listTemplatesXHR = props.jem.listTemplates((error, credentials) => {
+        this._abortXHR('_listTemplatesXHR');
+        if (error) {
+          console.error('Unable to list templates', error);
+          return;
+        }
+        // The credentials are returned in an array so we need to loop through
+        // the list to find the one with the proper path.
+        credentials.some(credential => {
+          if (credential.path === templateName && credential.location) {
+            deploymentStorage.cloud = credential.location.cloud;
+            deploymentStorage.region = credential.location.region;
+            return true;
+          }
+        });
+        if (deploymentStorage.cloud && deploymentStorage.region) {
+          this._listRegionsXHR = props.jem.listRegions(
+            deploymentStorage.cloud, (error, regions) => {
+              this._abortXHR('_listRegionsXHR');
```

Comment:
Technically we don't need to abort it, but this just cleans up after the request by also nulling out the reference to it.
I wanted to keep the requests separate for now so that we can track them individually in the future if needed. I don't actually like that we're making these requests her at all. We should be requesting this from the DB and have the DB fetch them for us. But that is in the pipeline for sometime in the future.